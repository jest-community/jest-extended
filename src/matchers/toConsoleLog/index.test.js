import matcher from './';

expect.extend(matcher);

const log = () => console.log('console-log'); // eslint-disable-line
const error = () => console.error('console-error'); // eslint-disable-line
const info = () => console.info('console-info'); // eslint-disable-line
const noop = () => {};

describe('.toConsoleLog', () => {
  test('passes when the callback function outputs the expected message to the console', () => {
    expect(log).toConsoleLog('console-log');
    expect(() => console.log(1)).toConsoleLog(1); // eslint-disable-line
    expect(() => console.log([1, 2])).toConsoleLog([1, 2]); // eslint-disable-line
    expect(() => console.log({ a: 1, b: 2 })).toConsoleLog({ a: 1, b: 2 }); // eslint-disable-line
  });
  test('fails when the callback function does not output any message', () => {
    expect(() => expect(noop).toConsoleLog('something')).toThrowErrorMatchingSnapshot();
  });
  test('fails when the callback function does not output the expected message', () => {
    expect(() => expect(log).toConsoleLog('something')).toThrowErrorMatchingSnapshot();
    expect(() =>
      expect(() => {
        console.log('a', 'b', 'c'); // eslint-disable-line
        log();
      }).toConsoleLog('something')
    ).toThrowErrorMatchingSnapshot();
  });
  test('passes when the callback function outputs the expected message among others', () => {
    expect(() => {
      console.log('something'); // eslint-disable-line
      log();
    }).toConsoleLog('console-log');
  });
});

describe('.toConsoleInfo', () => {
  test('passes when the callback function outputs the expected message to the console', () => {
    expect(info).toConsoleInfo('console-info', 'info');
  });
  test('fails when the callback function does not output any message', () => {
    expect(() => expect(noop).toConsoleInfo('something', 'info')).toThrowErrorMatchingSnapshot();
  });
  test('fails when the callback function does not output the expected message', () => {
    expect(() => expect(info).toConsoleInfo('something', 'info')).toThrowErrorMatchingSnapshot();
  });
});

describe('.toConsoleError', () => {
  test('passes when the callback function outputs the expected message to the console', () => {
    expect(error).toConsoleError('console-error', 'error');
  });
  test('fails when the callback function does not output any message', () => {
    expect(() => expect(noop).toConsoleError('something', 'error')).toThrowErrorMatchingSnapshot();
  });
  test('fails when the callback function does not output the expected message', () => {
    expect(() => expect(error).toConsoleError('something', 'error')).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toConsoleLog', () => {
  test('passes when the callback function does not output anything', () => {
    expect(noop).not.toConsoleLog('something');
  });
  test('passes when the callback function does not output the expected message', () => {
    expect(log).not.toConsoleLog('something');
  });
  test('fails when the callback function outputs the expected message', () => {
    expect(() => expect(log).not.toConsoleLog('console-log')).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toConsoleInfo', () => {
  test('passes when the callback function does not output anything', () => {
    expect(noop).not.toConsoleInfo('something', 'info');
  });
  test('passes when the callback function does not output the expected message', () => {
    expect(info).not.toConsoleInfo('something', 'info');
  });
  test('fails when the callback function outputs the expected message', () => {
    expect(() => expect(info).not.toConsoleInfo('console-info', 'info')).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toConsoleError', () => {
  test('passes when the callback function does not output anything', () => {
    expect(noop).not.toConsoleError('something', 'error');
  });
  test('passes when the callback function does not output the expected message', () => {
    expect(error).not.toConsoleError('something', 'error');
  });
  test('fails when the callback function outputs the expected message', () => {
    expect(() => expect(error).not.toConsoleError('console-error', 'error')).toThrowErrorMatchingSnapshot();
  });
});
