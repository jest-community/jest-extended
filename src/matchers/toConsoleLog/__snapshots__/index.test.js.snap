// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toConsoleError fails when the callback function outputs the expected message 1`] = `
"<dim>expect(</><red>received</><dim>).not.toConsoleError(</><green>expected</><dim>)</>

Expected console.error not to have been called with:
  <green>\\"console-error\\"</>
But it was called with:
  <red>[[\\"console-error\\"]]</>"
`;

exports[`.not.toConsoleInfo fails when the callback function outputs the expected message 1`] = `
"<dim>expect(</><red>received</><dim>).not.toConsoleInfo(</><green>expected</><dim>)</>

Expected console.info not to have been called with:
  <green>\\"console-info\\"</>
But it was called with:
  <red>[[\\"console-info\\"]]</>"
`;

exports[`.not.toConsoleLog fails when the callback function outputs the expected message 1`] = `
"<dim>expect(</><red>received</><dim>).not.toConsoleLog(</><green>expected</><dim>)</>

Expected console.log not to have been called with:
  <green>\\"console-log\\"</>
But it was called with:
  <red>[[\\"console-log\\"]]</>"
`;

exports[`.toConsoleError fails when the callback function does not output any message 1`] = `
"<dim>expect(</><red>received</><dim>).not.toConsoleError(</><green>expected</><dim>)</>

Expected console.error to have been called with:
  <green>\\"something\\"</>
But it was not called"
`;

exports[`.toConsoleError fails when the callback function does not output the expected message 1`] = `
"<dim>expect(</><red>received</><dim>).not.toConsoleError(</><green>expected</><dim>)</>

Expected console.error to have been called with:
  <green>\\"something\\"</>
But it was called with:
  <red>[[\\"console-error\\"]]</>"
`;

exports[`.toConsoleInfo fails when the callback function does not output any message 1`] = `
"<dim>expect(</><red>received</><dim>).toConsoleInfo(</><green>expected</><dim>)</>

Expected console.info to have been called with:
  <green>\\"something\\"</>
But it was not called"
`;

exports[`.toConsoleInfo fails when the callback function does not output the expected message 1`] = `
"<dim>expect(</><red>received</><dim>).toConsoleInfo(</><green>expected</><dim>)</>

Expected console.info to have been called with:
  <green>\\"something\\"</>
But it was called with:
  <red>[[\\"console-info\\"]]</>"
`;

exports[`.toConsoleLog fails when the callback function does not output any message 1`] = `
"<dim>expect(</><red>received</><dim>).toConsoleLog(</><green>expected</><dim>)</>

Expected console.log to have been called with:
  <green>\\"something\\"</>
But it was not called"
`;

exports[`.toConsoleLog fails when the callback function does not output the expected message 1`] = `
"<dim>expect(</><red>received</><dim>).toConsoleLog(</><green>expected</><dim>)</>

Expected console.log to have been called with:
  <green>\\"something\\"</>
But it was called with:
  <red>[[\\"console-log\\"]]</>"
`;

exports[`.toConsoleLog fails when the callback function does not output the expected message 2`] = `
"<dim>expect(</><red>received</><dim>).toConsoleLog(</><green>expected</><dim>)</>

Expected console.log to have been called with:
  <green>\\"something\\"</>
But it was called with:
  <red>[[\\"a\\", \\"b\\", \\"c\\"], [\\"console-log\\"]]</>"
`;
